### ISSUE TITLE
Cannot reproduce paper results

### ISSUE THREAD

**Issue by HuiHuiSun** [cshizhe/hiveformer#2]
Hi Pierre-Louis,

The idea of this paper is great, I am trying to reproduce the results in your paper.
However, I have encountered some problems in reproducing the single-task learning in Table 2 of your paper. 
I have only trained three single-tasks, they are Push Button, Put Knife, and Take Umbrella, in which the success rate of Push Button is the same as in the paper, but the success rate of Put Knife is 59.6%  , and the success rate of Take Umbrella is 33.4%, which is quite different from Table 2. 
The training parameters and dataset are from the readme.md, I only modified the path and changed the batchsize to 10.
Am I missing something?

**Comment by cshizhe** [comment]
Hi,

Since there is quite some randomness inside the RLBench, the generated data and evaluation episodes can be different across machines even with the same seed.
We suggest using our provided singularity image and run experiments for three times to get more stable results.

Just for your information,  the results in the single-task setup with our singularity image and datasets are presented below. The average preformance is similar to what reported in the paper:


|                                | pick_and_lift | pick_up_cup | put_knife | put_money_in_safe | push_button | reach_target | slide_block_to_target | stack_wine | take_money | take_umbrella | Avg.  |
|--------------------------------|---------------|-------------|-----------|-------------------|-------------|--------------|-----------------------|------------|------------|---------------|-------|
| Hiveformer reported            |         92.20 |       77.10 |     69.70 |             96.20 |       99.60 |       100.00 |                 95.40 |      81.90 |      82.10 |         90.10 | 88.43 |
| Results obtained from the repo |         92.64 |       89.76 |     75.76 |             95.66 |       91.25 |       100.00 |                 97.55 |      91.92 |      73.48 |         53.20 | 86.12 |

**Comment by HuiHuiSun** [comment]
Thank you very much.
I have successfully run the training code using singularity image. 
However, when I run the evaluation code using the singularity image, I get an error that I can't resolve, it seems that I can't open CoppeliaSim using the image python env, but if I don't use the python in the image,  I can open CoppeliaSim normally with my python env. 
Have you guys ever encountered the same problem when using the singularity image.

The following is the error encountered when running the evaluation program using the image file：

Error: signal 11:
lopt/Coppeliasim Edu V4 1 0 Ubuntu20 04/libcoppeliasim.so.1( Z11 segHandleri+0x30)[0x7f60835adae0]
/lib/x86 64-linux-gnu/libc.so.6(+0x42520)[0x7f6162504520]
"honesun/software/coppeliasin EdU V4 1 0 Ubuntu20 04/libcopeliasin,so( Z29sincetint32Paraneter internalipi+0x2da)(0x7f602e52a68a
/home/sun/software/Coppeliasim Edu V4 1
0 Ubuntu20 04/libsimExtCustomuI.so( Z21simGetInt32ParameterEi+0x29)[0x7f602501af19]
home/sun/software/Coppeliasim Edu V4 1
0 Ubuntu20 04/libsimExtCustomui.so(simstart+oxdc)[ox7f6024ff46ac7
/opt/Coppeliasim Edu V4 1 0 Ubuntu20 04/libcoppeliasim.so.1
ZN7CPlugin4loadEv+oxa7)[ox7f6083775e17
lopt/Coppeliasin Edu V4 1 0 Ubuntu20 04/libcopeliasin.so.1( ZN16CPlugincontainer9addpluginEPKCS1 +0x3df)(0x7f608377deaf)
ropt/copeliasin EdU V4 1 0 Ubuntu20 04/liboppeliasin.s.1 ZN9UThread27 executeormandvialithreadEp16SuThreadomands1 +0x247a)(0X7f60370d79a
opt/copeliasin Ed V4 1 0 buntu20 04/libcoppeltasin.so.1( ZN9CiThread2executeomnand/iaithreadEP16SuThreadConnands1 +0x3d)0x7f608370e19d
/opt/Coppeliasim_Edu_V4 1_0 Ubuntu20_04/libcoppeliasim.so.1( Z22simloadModule_internalpKcs0+0x3fc)[0x7f60835b172cJ

QMutex: destroying locked mutex

**Comment by cshizhe** [comment]
It seems that the problem is you didn't set up the correct path for Coppeliasim in the singlarity container. Please follow this to setup the path: https://github.com/stepjam/PyRep#install. It should be in the /opt/ directory.
